### PR TITLE
[backend] feature(freetrials): autoregister endpoint #1191

### PR DESCRIPTION
**Pull request by carinelebas**
# Context
- Create a new endpoint to autoregister a platform
- create a new platform_token directive to authenticate on that route. We restrict the permission to authorize platform registration on the targetted orga only. 

## How to test
- test the endpoint in apollo sandbox

## Related issues

- Related to #1191

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/security/directive-graphql/validators/platform-token-validator.test.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+      const result = await validateAndGetRequestedPlatformToken(req);
+
+      expect(result).toBe(null);
+    });
+
+    it('should return null when platform id header is missing', async () => {
+      const req: express.Request = {
+        headers: {
+          [PLATFORM_TOKEN_HEADER]: 'anything',
```
Comment (hervyt) [inline apps/portal-api/src/security/directive-graphql/validators/platform-token-validator.test.ts]: You talk about id... But you do about token..... 🤔 (same as the switch in the function)

File: apps/portal-api/src/modules/services/registration/registration.app.test.ts [hunk 2]
```diff
@@ -722,4 +740,115 @@ describe('Registration app', () => {
...
     });
   });
+
+  describe('autoRegisterPlatform', () => {
+    let deploymentRequest: DeploymentRequest;
+    const platformConfiguration = {
+      id: uuidv4(),
+      title: 'My OpenCTI platform',
+      url: 'http://example.com',
+      contract: PlatformContract.Ee,
```
Comment (hervyt) [inline apps/portal-api/src/modules/services/registration/registration.app.test.ts]: **question**: For now, should be Trial only the auto registration, no ? :)

File: apps/portal-api/src/security/directive-graphql/validators/platform-token-validator.test.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import { DeploymentRequestDomain } from '../../../modules/services/deployments/deployments.domain';
+import {
+  PLATFORM_ID_HEADER,
+  PLATFORM_TOKEN_HEADER,
+  validateActivePlatformToken,
+  validateAndGetRequestedPlatformToken,
+} from './platform-token-validator';
+
+describe('Platform Token Validation', () => {
+  describe('validateActivePlatformToken', () => {
```
Comment (Kroustille) [inline apps/portal-api/src/security/directive-graphql/validators/platform-token-validator.test.ts]: I think we're missing a test where an active configuration is not found.

File: apps/portal-api/src/security/directive-graphql/validators/platform-token-validator.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  ) {
+    const isActivePlatformToken = await validateActivePlatformToken(
+      portalContext.req
+    );
+
+    const deploymentRequest = await validateAndGetRequestedPlatformToken(
+      portalContext.req
+    );
+
+    if (!isActivePlatformToken && !deploymentRequest) {
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/security/directive-graphql/validators/platform-token-validator.ts]: well spotted, fixed in [fa1dd51](https://github.com/FiligranHQ/xtm-hub/pull/1202/commits/fa1dd519c8fd9958483cafe47864a63b6f3de658)
